Add an option to drain child process output before termination (#8434)
alacritty/alacritty: alacritty_terminal/CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

## 0.25.0-dev

### Changed

> - Replaced `Options::hold` with `Options::drain_on_exit` that drains the child process output but does not hold the terminal open. Hold should now be implemented by the user.

Reviewer: You should rename the `0.24.3-dev` to `0.25.0-dev`.

> - Replaced `Options::hold` with `Options::drain_on_exit` that drains the child process output but does not hold the terminal open. Hold should now be implemented by the user.

Reviewer: ```suggestion
## 0.25.0-dev

### Changed

- Replaced `Options::hold` with `Options::drain_on_exit` that drains, but doesn't hold, since holding could be easily done by user.
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

## 0.25.0-dev

### Changed

> - Replaced `Options::hold` with `Options::drain_on_exit` that drains, but doesn't hold, since holding could be easily done by user

Reviewer 2: ```suggestion
- Replaced `Options::hold` with `Options::drain_on_exit` that drains, but doesn't hold, since holding can be done outside of alacritty_terminal
```

"could" and "easily" are two words I'd rather not use in a changelog since they're vague and superfluous. Additionally "user" is kinda confusing because this is about the library consumer not the terminal user.
